File: 2315-count-asterisks/2315-count-asterisks.java
class Solution {
    public int countAsterisks(String s) {
        int ast = 0;
        int bar = 0;

        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '|') 
                bar++;
            else if(bar % 2 == 0){
                if(s.charAt(i) == '*')
                    ast++;
            }
        }
        return ast;
    }
}

File: 2315-count-asterisks/2315-count-asterisks.swift
class Solution {
    func countAsterisks(_ s: String) -> Int {
        var pipes = 0
        var ast = 0
        
        for c in s {
            if c == "|" {
                pipes += 1
            } else if pipes % 2 == 0 {
                if c == "*" {
                    ast += 1
                }
            }
        }
        return ast
    }
}